docs/howto/customization/mapping_parser.md

```diff
@@ -1,12 +1,130 @@
 # How to write data to archive with MappingParser
 
-`MappingParser` is a generic parser class implemented in
-`nomad.parsing.file_parser/mapping_parser.py` to handle the conversion to and from a
+`MappingParser` is a NOMAD parsing framework for handling tree-like data structures.
+It is extensively used for processing highly structured files, like computational output.
+This framework distributes responsiblities between the file processing (building the source tree) and the mapping into the NOMAD schema.
+Some functionality can be inserted in the mapping, the rest is relegated to *normalization* functionalitites of the schema.
+
+![Responsiblity distribution MappingParser](images/mapping_parser_concept.png)
+
+<!-- `nomad.parsing.file_parser/mapping_parser.py` to handle the conversion to and from a
 data object and a python dictionary. We refer to an instance of the
 this class as 'mapping parser' throughout this section. In the following, the abstract
 properties and methods of the mapping parser are explained. The various implementations of
 the mapping parser are also defined and `Mapper` which is required to convert a
-mapping parser into another mapping parser is explained as well.
+mapping parser into another mapping parser is explained as well. -->
+
+## Fundamentals
+
+The mapping from the source file to the archive is defined by a path (in [JMesPath](https://jmespath.org/) format) to the schema.
+This path is added to `m_annotations`, either overwriting or extending the previous annotations.
+Each mapping corresponds to its own dictionary key, and a parser schema may contain multiple in parallel.
```

Comment:
Maybe flip this, multiple annotations can be provided with their own key, which can be set prior to mapping.

Reply:
You mean that people provide just the keys, but not the values..? Is that really useful to mention?